domehunter/dome_control.py

```diff
@@ -319,31 +353,28 @@ def goto_az(self, az):
         """
         if self.dome_az is None:
             return
-
-        target_az = Longitude(az * u.deg)
-        logger.debug(f'Goto target azimuth: {target_az}')
+        if self._move_event.is_set():
+            logger.debug('Movement command in progress')
+            raise Exception("Movement command in progress")
+        # need target_az for _goto_az_condition but unsure how to pass it
+        # through the thread as an argument for _goto_az_condition.
+        # creating a instance variable as a work around.
+        self.target_az = Longitude(az * u.deg)
```

Comment:
An instance variable is OK (though probably should be private), but if you want to to pass it around instead you could have `self._thread_condition` accept some additional args that it passes to the condition function, e.g.
```
def _thread_condition(self, trigger_condition, *args, **kwargs):

    ...

    if trigger_condition(*args, **kwargs):

    ...

```
Then the `trigger_condition` methods can have whatever arguments you want, and you just pass them though via `self._thread_condition` as additional arguments when you create the `Thread`, e.g.
```
goingto_az = threading.Thread(target=self._thead_condition,
                              args=(self._goto_az_condition,
                                    target_az))

...

def _goto_az_condition(self, target_az):

    ...

```